File: src/main/java/org/distril/beengine/scheduler/task/TaskEntry.java
package org.distril.beengine.scheduler.task;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;

@Getter
@RequiredArgsConstructor
public class TaskEntry {

	private final Task task;
	private final int delay, period;
	private final boolean async;
	@Setter
	private long nextRunTick;

	public boolean isCancelled() {
		return this.task.isCancelled();
	}

	public boolean isRepeating() {
		return this.period > 0;
	}
}

File: src/main/kotlin/org/distril/beengine/scheduler/task/TaskEntry.kt
package org.distril.beengine.scheduler.task

class TaskEntry(
	val task: Task,
	val delay: Int,
	val period: Int,
	val isAsync: Boolean,
	var nextRunTick: Long = 0
) : Comparable<TaskEntry> {

	var isCancelled = false
		private set

	val isRepeating = this.period > 0

	fun cancel() {
		this.isCancelled = true

		this.task.onCancel()
	}

	override fun compareTo(other: TaskEntry) = this.nextRunTick.compareTo(other.nextRunTick)
}
